stores/src/main/java/dataBase/DataBaseStorePopulator.java

```diff
@@ -85,16 +43,72 @@
                 throw new RuntimeException("Database access error", se2);
             }
             try {
-                if (dataBaseConnect.connection() != null) dataBaseConnect.connection().close();
+                if (connection() != null) connection().close();
             } catch (SQLException se) {
-                throw new RuntimeException("Database access error", se);
+                throw new RuntimeException("Connection error", se);
             }
         }
-//        System.out.println(categoryListDB);
         return categoryListDB;
     }
 
-    private void fillCategoryDB() throws SQLException {
+    private Connection connection() {
+        String driver = "org.h2.Driver";
+        String url = "jdbc:h2:tcp://localhost/~/test";
+        String user = "sa";
+        String password = "";
+        Connection connection;
+        try {
+            Class.forName(driver);
+            connection = DriverManager.getConnection(url, user, password);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Exception thrown while loading the class", e);
+        } catch (SQLException e) {
+            throw new RuntimeException("Database access error", e);
+        }
+        return connection;
+    }
+
+    private void createTable() throws SQLException {
+        statement = connection().createStatement();
+        String sql1 = "CREATE TABLE IF NOT EXISTS CATEGORY " +
+                "(id INTEGER not NULL AUTO_INCREMENT," +
+                "name VARCHAR(50) NOT NULL, " +
+                " PRIMARY KEY ( id ))";
+        statement.executeUpdate(sql1);
+        String sql2 = "CREATE TABLE IF NOT EXISTS PRODUCT " +
+                "(id INTEGER not NULL AUTO_INCREMENT, " +
+                " productName VARCHAR(50), " +
+                " productPrice DOUBLE, " +
+                " productRate DOUBLE, " +
+                " category_id LONG NOT NULL, " +
+                " FOREIGN KEY (category_id) REFERENCES CATEGORY (id), " +
+                " PRIMARY KEY ( id ))";
+        statement.executeUpdate(sql2);
+    }
+
+    private void setDataTable() throws SQLException {
```

Comment:
why setDataTable? in this method you insert data to tables, so... insertDataToTables or similar